### PR TITLE
Rename e2e -> integration_test

### PR DESCRIPTION
**Pull request by dnfield**
The e2e name is not entirely fitting for this plugin - some of our consumers expect e2e to implement larger multi-system tests exercising both client and server, which makes this package confusing. Adding the word test also should make this package more discoverable on pub for users looking for test packages.

WIP while I figure out which other plugin(s) need bumps to accomodate pubspec changes.

### CODE REVIEW

**Comment by dnfield** [comment]
The lint failure is because it's looking for a URL that doesn't exist yet, since this PR will be creating that URL.

File: packages/integration_test/android/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-  package="dev.flutter.e2e">
+  package="dev.flutter.integration_test">
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/integration_test/android/src/main/AndroidManifest.xml:2]: Ahh good catch, I'll make this match up.

**Comment by mklim** [comment]
I don't see an obvious way that the previously existing e2e package on pub would be linked to the new name from this PR, is that a concern? I think it would probably help to keep e2e as an empty package in the repo and publish a README update about the rename to pub, but I'm not really familiar with all the context here.

**Comment by dnfield** [comment]
That's a good idea.  I was going to just locally modify e2e for that

File: packages/integration_test/lib/integration_test_driver.dart [hunk 2]
```diff
@@ -8,11 +8,11 @@ import 'dart:io';
 
 import 'package:flutter_driver/flutter_driver.dart';
 
-import 'package:e2e/common.dart' as e2e;
+import 'package:integration_test/common.dart';
 import 'package:path/path.dart' as path;
 
 /// This method remains for backword compatibility.
-Future<void> main() => e2eDriver();
+Future<void> integration_main() => integrationDriver();
```
Comment (CareF) [inline packages/integration_test/lib/integration_test_driver.dart:15]: If we are not calling it `main` anyway, why not just remove it? I was thinking if it's `main` it automatically can be used as a driver script.

**Comment by CareF** [comment]
And there's `readme.md` in root directory.

**Comment by dnfield** [comment]
@CareF I've updated the READMe. I'm not calling it main so you can import it without an alias more easily and still use it.

**Comment by CareF** [comment]
> I'm not calling it main so you can import it without an alias more easily and still use it.

I get that, but I mean `Future<void> integration_main() => integrationDriver();` this is no longer necessary if we don't want a `main` here. Users can directly use `integrationDriver` in their driver file.

**Comment by dnfield** [comment]
This is currently blocked by apparently unrelated timeouts on infra. Trying to open another PR to shard.

**Comment by dnfield** [comment]
Red CI is because this PR creates a directory the CI is checking for.

Landing on red for that reason. If this goes red on post submit will revert.

**Pull request merged**